static/js/components/dashboard/CourseAction_test.js

```diff
@@ -5,34 +5,65 @@ import { shallow } from 'enzyme';
 import moment from 'moment';
 import { assert } from 'chai';
 import sinon from 'sinon';
+import _ from 'lodash';
 
 import CourseAction from './CourseAction';
 import {
   DASHBOARD_FORMAT,
+  COURSE_PRICES_RESPONSE,
+  FINANCIAL_AID_PARTIAL_RESPONSE,
   STATUS_PASSED,
   STATUS_NOT_PASSED,
   STATUS_OFFERED,
   STATUS_CAN_UPGRADE,
   STATUS_CURRENTLY_ENROLLED,
+  FA_PENDING_STATUSES,
+  FA_STATUS_SKIPPED
 } from '../../constants';
 import { findCourse } from './CourseDescription_test';
 
 describe('CourseAction', () => {
   const now = moment();
-  let sandbox, checkoutStub;
+  let sandbox, checkoutStub, coursePrice, defaultParams, defaultParamsNow;
 
   beforeEach(() => {
     sandbox = sinon.sandbox.create();
     checkoutStub = sandbox.stub();
+    coursePrice = COURSE_PRICES_RESPONSE[1];
+    defaultParams = {
+      checkout: checkoutStub,
+      coursePrice: coursePrice
+    };
+    defaultParamsNow = Object.assign({}, defaultParams, { now: now });
   });
 
   afterEach(() => {
     sandbox.restore();
   });
 
+  let getElements = (renderedComponent) => {
+    let buttonContainer = renderedComponent.find(".course-action-action");
+    let button = buttonContainer.find(".dashboard-button");
+    let buttonText;
+    if (!_.isNil(button) && button.children().length > 0) {
+      buttonText = button.children().text();
+    }
+    let description = renderedComponent.find(".course-action-description");
+    return {
+      buttonContainer: buttonContainer,
+      button: button,
+      buttonText: buttonText,
+      description: description
+    };
+  };
+
+  let alterFirstRun = (course, overrideObject) => {
+    course.runs[0] = Object.assign({}, course.runs[0], overrideObject);
```

Comment:
i'm actually trying to modify the object in place.  `CourseAction` takes a `course` object, and `course` is a complex object. the goal here is just to alter some data on the first run. I realize this is a side effect and not the best pattern. would you rather i change this to return a copy of `course` with the modified first run? if so, this function would be more accurately called `courseWithAlteredFirstRun` or something like that